config/src/lib.rs

```diff
@@ -901,6 +903,95 @@ impl Config {
 
         Ok(())
     }
+
+    /// List the data in the foundry cache
+    pub fn list_foundry_cache() -> eyre::Result<Cache> {
+        if let Some(cache_dir) = Config::foundry_cache_dir() {
+            let mut cache = Cache { chains: vec![] };
+            if let Ok(entries) = cache_dir.as_path().read_dir() {
+                for entry in entries.flatten() {
+                    if let Ok(blocks) = Self::get_cached_blocks(&entry.path()) {
+                        cache.chains.push(ChainCache {
+                            name: entry.file_name().into_string().unwrap(),
+                            blocks,
+                        })
+                    }
+                }
+                Ok(cache)
+            } else {
+                eyre::bail!("failed to access foundry_cache_dir");
+            }
+        } else {
+            eyre::bail!("failed to get foundry_cache_dir");
+        }
+    }
+
+    /// List the cached data for `chain`
+    pub fn list_foundry_chain_cache(chain: Chain) -> eyre::Result<ChainCache> {
+        if let Some(cache_dir) = Config::foundry_chain_cache_dir(chain) {
+            let blocks = Self::get_cached_blocks(&cache_dir)?;
+            Ok(ChainCache { name: chain.to_string(), blocks })
+        } else {
+            eyre::bail!("failed to get foundry_chain_cache_dir");
+        }
+    }
+
+    fn get_cached_blocks(chain_path: &Path) -> eyre::Result<Vec<(String, u64)>> {
+        if let Ok(chain_dir) = chain_path.read_dir() {
+            let mut blocks = vec![];
+            for block in chain_dir.flatten() {
```

Comment:
should be able to check the file type and filter only dirs, instead of attempting to read it as dir

Reply:
Same as the other comment about this, at this point we only care about dirs so we can reduce nesting by removing the `is_dir()` check and file will be ignored by `if let Ok()`

Reply:
this will be ok if we ensure that we only pass dirs here, but this seems unnecessary:

```rust
if let Ok(mut block_dir) = block.path().read_dir() {
                    let size = block_dir.next().unwrap()?.metadata()?.len();
                    blocks.push((block.file_name().to_string_lossy().into_owned(), size));
                }
```

if you could directly read its metadata